UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -2148,9 +2177,200 @@ private void MenuItem_Add_Click(object sender, RoutedEventArgs e)
             OpenInTab(obj, true);
         }
 
-        private void RootMenuItem_SubmenuOpened(object sender, RoutedEventArgs e)
+        private void MenuItem_CollapseAll_Click(object sender, RoutedEventArgs e)
         {
-            MenuItem_RunScript_SubmenuOpened(sender, e, Path.Combine(ExePath, "Scripts"));
+            foreach (TreeViewItem item in (MainTree.Items[0] as TreeViewItem).Items)
+                item.IsExpanded = false;
+        }
+
+        private void MenuItem_CollapseCurrent_Click(object sender, RoutedEventArgs e)
+        {
+            foreach (TreeViewItem item in (MainTree.Items[0] as TreeViewItem).Items)
+                if (item.Items.Contains(MainTree.SelectedItem))
+                    item.IsExpanded = false;
+        }
+
+        private void MenuItem_ContextMenuOpened(object sender, RoutedEventArgs e)
+        {
+            var menu = sender as ContextMenu;
+            int countOfItemsInMenu = MainTree.Items.Count;
+            //LMM: Regarding the text changes, and visibility changes here... there is most likely better ways to do this, I'm not super familair with XAML though, and eh, this worked for me.
+            //other contributors could polish this up if need be :)
```

Comment:
code style: please have a space after the `//` and start with an uppercase letter

What does `LMM` mean?

Also CC @VladiStep cause you probably know more xaml than anyone else on the dev team.